lib/experience-version.js

```diff
@@ -80,33 +81,80 @@ const choices = (nameField, versionField, datas) => {
   return mapping;
 };
 
+const promptForSlugsAndDomains = async (currentDomains, currentSlugs) => {
+  const domainChoiceMappings = choices('domainName', 'version', currentDomains);
+  const domainChoices = keys(domainChoiceMappings);
+  const slugChoiceMappings = choices('slug', 'version', currentSlugs);
+  const slugChoices = keys(slugChoiceMappings);
+  const questions = [];
+  if (domainChoices.length) {
+    questions.push({ type: 'checkbox', name: 'domains', message: 'Select Experience Domains to point at this version', choices: domainChoices });
+  }
+  if (slugChoices.length) {
+    questions.push({ type: 'checkbox', name: 'slugs', message: 'Select Experience Slugs to point at this version', choices: slugChoices });
+  }
+  let domainIds = [], slugIds = [];
+  if (questions.length) {
+    const { domains, slugs } = await inquirer.prompt(questions) || {};
+    if (domains?.length) {
+      domainIds = domains.map((domain) => domainChoiceMappings[domain]);
+    }
+    if (slugs?.length) {
+      slugIds = slugs.map((slug) => slugChoiceMappings[slug]);
+    }
+  }
+  return { domainIds, slugIds };
+};
+
 module.exports = async (version, opts = {}) => {
   const { apiToken, applicationId, api, endpointDomain } = await loadConfig();
   if (!applicationId || !apiToken) { return; } // config did not lock or did not load correctly.
   if (!version) {
     await listVersions(api, applicationId, opts.list, endpointDomain);
   } else {
-    const domainChoiceMappings = choices('domainName', 'version', await getExperiencePart(api, 'domain', { applicationId }));
-    const domainChoices = keys(domainChoiceMappings);
-    const slugChoiceMappings = choices('slug', 'version', await getExperiencePart(api, 'slug', { applicationId }));
-    const slugChoices = keys(slugChoiceMappings);
-    const questions = [];
-    if (domainChoices.length) {
-      questions.push({ type: 'checkbox', name: 'domains', message: 'Select Experience Domains to point at this version', choices: domainChoices });
-    }
-    if (slugChoices.length) {
-      questions.push({ type: 'checkbox', name: 'slugs', message: 'Select Experience Slugs to point at this version', choices: slugChoices });
-    }
-    let domainIds, slugIds;
-    if (questions.length) {
-      const { domains, slugs } = await inquirer.prompt(questions) || {};
-      if (domains?.length) {
-        domainIds = domains.map((domain) => domainChoiceMappings[domain]);
-      }
-      if (slugs?.length) {
-        slugIds = slugs.map((slug) => slugChoiceMappings[slug]);
+    const currentDomains = await getExperiencePart(api, 'domain', { applicationId });
+    const currentSlugs = await getExperiencePart(api, 'slug', { applicationId });
+    const versionInfo = { description: opts.description };
+    if (isNotNil(opts.slugIds) || isNotNil(opts.domainIds)) {
+      const domainAndSlugMap = new Map();
```

Comment:
I think keeping two maps would be best - since you are having the user pass them in as two params